File: src/main/java/jukebot/commands/Play.java
package jukebot.commands;

import jukebot.JukeBot;
import jukebot.audioutilities.AudioHandler;
import jukebot.audioutilities.SongResultHandler;
import jukebot.utils.Command;
import jukebot.utils.CommandProperties;
import jukebot.utils.Context;
import net.dv8tion.jda.core.managers.AudioManager;

@CommandProperties(description = "Finds and plays the provided song query/URL", aliases = {"p"}, category = CommandProperties.category.CONTROLS)
public class Play implements Command {

    public void execute(final Context context) {

        if (context.getArgString().isEmpty()) {
            context.embed("Play", "Specify a URL or a search term");
            return;
        }

        final AudioManager manager = context.getGuild().getAudioManager();
        final AudioHandler player = context.getAudioPlayer();

        final boolean voiceConnected = context.ensureVoice();

        if (!voiceConnected) {
            return;
        }

        if (!player.isPlaying()) {
            player.setChannel(context.getChannel().getIdLong());
        }

        final String userQuery = context.getArgString().replaceAll("[<>]", "");

        if (userQuery.startsWith("http")) {
            if (userQuery.toLowerCase().contains("/you/likes")) {
                context.embed("SoundCloud Liked Tracks", "JukeBot doesn't implement oauth and as a result\ncannot access your liked tracks when referenced as `you`");

                if (!player.isPlaying()) {
                    manager.closeAudioConnection();
                }
                return;
            }
            if (userQuery.toLowerCase().contains("pornhub") && !context.getChannel().isNSFW()) {
                context.embed("PornHub Tracks", "PornHub tracks can only be loaded from NSFW channels!");

                if (!player.isPlaying()) {
                    manager.closeAudioConnection();
                }

                return;
            }
            JukeBot.playerManager.loadItem(userQuery, new SongResultHandler(context, player, false));
        } else {
            JukeBot.playerManager.loadItem("ytsearch:" + userQuery, new SongResultHandler(context, player, false));
        }

    }
}

File: src/main/java/jukebot/commands/Play.kt
package jukebot.commands

import jukebot.JukeBot
import jukebot.audio.AudioHandler
import jukebot.audio.SongResultHandler
import jukebot.framework.Command
import jukebot.framework.CommandCategory
import jukebot.framework.CommandProperties
import jukebot.framework.Context

@CommandProperties(description = "Finds and plays the provided song query/URL", aliases = ["p"], category = CommandCategory.PLAYBACK)
class Play : Command(ExecutionType.TRIGGER_CONNECT) {

    override fun execute(context: Context) {
        val player = context.getAudioPlayer()

        if (!player.isPlaying) {
            player.channelId = context.channel.idLong
        }

        if (context.message.attachments.size > 0) {
            return loadWithAttachment(context, player)
        }

        loadWithArgs(context, player)
    }

    fun loadWithAttachment(ctx: Context, player: AudioHandler) {
        JukeBot.playerManager.loadItem(ctx.message.attachments[0].url, SongResultHandler(ctx, player, false))
    }

    fun loadWithArgs(ctx: Context, player: AudioHandler) {
        val manager = ctx.guild.audioManager
        val userQuery = ctx.argString.replace("[<>]".toRegex(), "")

        if (userQuery.startsWith("http")) {
            if (userQuery.toLowerCase().contains("/you/likes")) {
                ctx.embed("SoundCloud Liked Tracks", "JukeBot doesn't implement oauth and as a result\ncannot access your liked tracks when referenced as `you`")

                if (!player.isPlaying) {
                    manager.closeAudioConnection()
                }
                return
            }
            if (userQuery.toLowerCase().contains("pornhub") && !ctx.channel.isNSFW) {
                ctx.embed("PornHub Tracks", "PornHub tracks can only be loaded from NSFW channels!")

                if (!player.isPlaying) {
                    manager.closeAudioConnection()
                }

                return
            }
            JukeBot.playerManager.loadItem(userQuery, SongResultHandler(ctx, player, false))
        } else {
            JukeBot.playerManager.loadItem("ytsearch:$userQuery", SongResultHandler(ctx, player, false))
        }
    }
}
